pkg/api/norman/customization/alibaba/handler.go

```diff
@@ -0,0 +1,370 @@
+			logrus.Debugf("[alibaba-handler] error call describeKeyPairs: %v", err)
+			util.ReturnHTTPError(writer, req, errCode, err.Error())
+			return
+		}
+		writer.Write(serialized)
+	case "alibabaVpcs":
+		if serialized, errCode, err = describeVpcs(capabilities, req); err != nil {
+			logrus.Debugf("[alibaba-handler] error call describeVpcs: %v", err)
+			util.ReturnHTTPError(writer, req, errCode, err.Error())
+			return
+		}
+		writer.Write(serialized)
+	case "alibabaVSwitches":
+		if serialized, errCode, err = describeVSwitches(capabilities, req); err != nil {
+			logrus.Debugf("[alibaba-handler] error call describeVSwitches: %v", err)
+			util.ReturnHTTPError(writer, req, errCode, err.Error())
+			return
+		}
+		writer.Write(serialized)
+	case "alibabaAvailableResources":
+		if serialized, errCode, err = describeAvailableResource(capabilities, req); err != nil {
+			logrus.Debugf("[alibaba-handler] error call describeAvailableResource: %v", err)
+			util.ReturnHTTPError(writer, req, errCode, err.Error())
+			return
+		}
+		writer.Write(serialized)
+	case "alibabaKubernetesVersions":
+		if serialized, errCode, err = describeKubernetesMetadata(capabilities, req); err != nil {
+			logrus.Debugf("[alibaba-handler] error call describeKubernetesMetadata: %v", err)
+			util.ReturnHTTPError(writer, req, errCode, err.Error())
+			return
+		}
+		writer.Write(serialized)
+	case "alibabaImageSupportedInstanceTypes":
+		if serialized, errCode, err = describeImageSupportedInstanceTypes(capabilities, req); err != nil {
+			logrus.Debugf("[alibaba-handler] error call describeImageSupportedInstanceTypes: %v", err)
+			util.ReturnHTTPError(writer, req, errCode, err.Error())
+			return
+		}
+		writer.Write(serialized)
+	default:
+		handleErr(writer, httperror.NotFound.Status, fmt.Errorf("invalid endpoint %v", resourceType))
+	}
+}
+
+func (h *handler) checkCredentials(req *http.Request) (int, error) {
+	cred := &Capabilities{}
+	raw, err := io.ReadAll(req.Body)
+	if err != nil {
+		return http.StatusBadRequest, fmt.Errorf("cannot read request body: %v", err)
+	}
+
+	if err = json.Unmarshal(raw, &cred); err != nil {
+		return http.StatusBadRequest, fmt.Errorf("cannot parse request body: %v", err)
+	}
+
+	if cred.RegionId == "" {
+		cred.RegionId = defaultRegion
+	}
+	if cred.Ak == "" {
+		return http.StatusBadRequest, fmt.Errorf("must provide access key ID")
+	}
+	if cred.Sk == "" {
+		return http.StatusBadRequest, fmt.Errorf("must provide access key secret")
+	}
+
+	client, err := CreateECSClient(cred.Ak, cred.Sk, cred.RegionId)
+	if err != nil {
+		return http.StatusInternalServerError, err
+	}
+
+	request := &ecs.DescribeRegionsRequest{}
+	if cred.AcceptLanguage != "" {
+		request.AcceptLanguage = &cred.AcceptLanguage
+	}
+
+	_, err = client.DescribeRegions(request)
+	if err != nil {
+
+		logrus.Debugf("[alibaba-handler] error call describeRegions: %v", err)
+		return handleSDKError(err)
+	}
```

Comment:
Is this used in order to validate the credentials? By making a real API call?

Reply:
yes